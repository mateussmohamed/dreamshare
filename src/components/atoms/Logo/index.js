import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { em } from 'polished'

const LogoStyled = styled.a`
  ${props => props.theme.typography.montserratStyle};
  font-size: ${props => em(props.size)};
  font-weight: bold;
  text-transform: uppercase;
  color: ${props => props.color};
`

const Logo = props => <LogoStyled {...props}>{props.text}</LogoStyled>

Logo.defaultProps = {
  text: 'Logo',
  color: '#fff',
  size: 20,
}

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.number,
}

export default Logo
