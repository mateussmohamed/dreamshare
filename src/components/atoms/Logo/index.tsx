import React from 'react'

import styled from 'styled-components'

interface LogoProps {
  text?: string
  color?: string
  size?: number
  weight?: string
}

const LogoStyled = styled.a<LogoProps>`
  ${props => props.theme.typography.montserratStyle};
  font-size: ${props => props.size};
  font-weight: ${props => props.weight};
  text-transform: uppercase;
  color: ${props => props.color};
`

export const Logo = (props: LogoProps) => {
  const { text = 'Logo', color = '#fff', size = 20, weight = 'bold' } = props

  return (
    <LogoStyled color={color} size={size} weight={weight}>
      {text}
    </LogoStyled>
  )
}
