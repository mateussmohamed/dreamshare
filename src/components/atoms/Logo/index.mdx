---
name: Logo
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import Logo from './index'
import LayoutTemplate from '../../templates/LayoutTemplate'

# Logo
<PropsTable of={Logo} />

## Basic usage
<Playground>
  <LayoutTemplate>
    <div style={{ background: '#bfbfbf', padding: 20 }}>
      <Logo />
      <br />
      <Logo text="Dreamshare" />
      <br />
      <Logo text="Dreamshare" color="tomato" />
      <br />
      <Logo text="Dreamshare" color="purple" size={40} />
    </div>
  </LayoutTemplate>
</Playground>
