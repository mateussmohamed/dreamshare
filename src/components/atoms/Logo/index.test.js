import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Logo from './index'

describe('[<Logo />]', () => {
  const wrap = (props = {}) => shallow(<Logo {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ children: 'I am a Title' })

    expect(wrapper).toMatchSnapshot()
  })
})
