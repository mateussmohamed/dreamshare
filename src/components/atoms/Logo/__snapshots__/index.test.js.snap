// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Logo />] matches to snapshot 1`] = `
<StyledComponent
  color="#fff"
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Logo__LogoStyled-sc-11yt4xf-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";font-size:",
          [Function],
          ";font-weight:bold;text-transform:uppercase;color:",
          [Function],
          ";",
        ],
      },
      "displayName": "Logo__LogoStyled",
      "render": [Function],
      "styledComponentId": "Logo__LogoStyled-sc-11yt4xf-0",
      "target": "a",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  size={20}
  text="Logo"
>
  Logo
</StyledComponent>
`;
